src/include/sof/wait.h

```diff
@@ -34,48 +34,17 @@
 #define __INCLUDE_WAIT__
 
 #include <stdint.h>
-#include <errno.h>
+
 #include <arch/wait.h>
-#include <sof/io.h>
-#include <sof/debug.h>
-#include <sof/work.h>
-#include <sof/timer.h>
-#include <sof/interrupt.h>
-#include <sof/trace.h>
-#include <sof/lock.h>
-#include <sof/clk.h>
-#include <platform/clk.h>
-#include <platform/interrupt.h>
-#include <sof/drivers/timer.h>
-#include <platform/platform.h>
-
-#if DEBUG_LOCKS
-#define wait_atomic_check	\
-	if (lock_dbg_atomic) { \
-		trace_error_atomic(TRACE_CLASS_WAIT, "atm"); \
-	}
-#else
-#define wait_atomic_check
-#endif
 
-#define DEFAULT_TRY_TIMES 8
+#include <sof/work.h>
 
 typedef struct {
 	uint32_t complete;
 	struct work work;
 	uint64_t timeout;
 } completion_t;
 
-void arch_wait_for_interrupt(int level);
-
-static inline void wait_for_interrupt(int level)
-{
-	tracev_event(TRACE_CLASS_WAIT, "WFE");
-	wait_atomic_check;
-	arch_wait_for_interrupt(level);
```

Comment:
please keep the small functions inline, this way we can avoid unnecessary calls and window exceptions on xtensa

Reply:
@lgirdwood should be better now